spring-data-cassandra/src/main/java/org/springframework/data/cassandra/core/ReactiveCassandraOperations.java

```diff
@@ -97,6 +99,19 @@
 	 */
 	<T> Flux<T> select(Statement statement, Class<T> entityClass) throws DataAccessException;
 
+	/**
+	 * Execute a {@code SELECT} query with paging and convert the result set to a {@link Slice} of entities.
+	 *
+	 * A sliced query translates the effective {@link Statement#getFetchSize() fetch size} to the page size.
+	 *
+	 * @param statement the CQL statement, must not be {@literal null}.
+	 * @param entityClass The entity type must not be {@literal null}.
+	 * @return the result object returned by the action or {@link Mono#empty()}
```

Comment:
In this case, we're mixing paradigms. We should return an empty `Slice` (see `Page.empty()`) as the query terminates without rows and the query result is the container for rows, not the rows themselves.